docs/reference/query-dsl/multi-match-query.asciidoc

```diff
@@ -247,273 +250,198 @@ number of `match` clauses.
-        { "match_phrase_prefix": { "message": "quick brown f" }}
-      ]
-    }
-  }
-}
---------------------------------------------------
+    +(combined("database", fields:["title", "abstract"]))
+    +(combined("systems", fields:["title", "abstract"]))
 
-Also, accepts `analyzer`, `boost`, `lenient` and `zero_terms_query` as explained
-in <<query-dsl-match-query>>, as well as `slop` which is explained in <<query-dsl-match-query-phrase>>.
-Type `phrase_prefix` additionally accepts `max_expansions`.
+In other words, *all terms* must be present *in at least one field* for a
+document to match.  (Compare this to <<operator-min,the logic used for
+`best_fields` and `most_fields`>>.)
+
+When scoring matches, the `combined_fields` type combines term and collection
+statistics across fields. This allows it to score each match as if the
+specified fields had been indexed into a single combined field. (Note that
+this is a best attempt -- `combined_fields` makes some approximations and
+scores will not obey this model perfectly.)
+
+This mode accepts the `analyzer`, `boost`, `operator`, `minimum_should_match`,
+`lenient` and `zero_terms_query` options.
 
 [IMPORTANT]
-[[phrase-fuzziness]]
-.`phrase`, `phrase_prefix` and `fuzziness`
+.`combined_fields` and `fuzziness`
 ===================================================
-The `fuzziness` parameter cannot be used with the `phrase` or `phrase_prefix` type.
+The `fuzziness` parameter cannot be used with the `combined_fields` or
+`cross_fields` types.
 ===================================================
 
-[[type-cross-fields]]
-==== `cross_fields`
-
-The `cross_fields` type is particularly useful with structured documents where
-multiple fields *should* match.  For instance, when querying the `first_name`
-and `last_name` fields for ``Will Smith'', the best match is likely to have
-``Will'' in one field and ``Smith'' in the other.
-
-****
-
-This sounds like a job for <<type-most-fields>> but there are two problems
-with that approach. The first problem is that `operator` and
-`minimum_should_match` are applied per-field, instead of per-term (see
-<<operator-min,explanation above>>).
-
-The second problem is to do with relevance: the different term frequencies in
-the `first_name` and `last_name` fields   can produce unexpected results.
-
-For instance, imagine we have two people: ``Will Smith'' and ``Smith Jones''.
-``Smith'' as a last name is very common (and so is of low importance) but
-``Smith'' as a first name is very uncommon (and so is of great importance).
-
-If we do a search for ``Will Smith'', the ``Smith Jones'' document will
-probably appear above the better matching ``Will Smith'' because the score of
-`first_name:smith` has trumped the combined scores of `first_name:will` plus
-`last_name:smith`.
-
-****
+[NOTE]
+.Custom similarities
+===================================================
+The `combined_fields` type only supports the `BM25` similarity, which is the
```

Comment:
We could technically support other similarities like `boolean` and `DFR`. But I opted to keep the message + implementation simple at first, especially since it's rare to have a custom similarity configuration.